vector/src/main/res/layout/activity_home.xml

```diff
@@ -30,74 +34,136 @@
         <ProgressBar
             android:id="@+id/home_recents_sync_in_progress"
             style="@style/Widget.AppCompat.ProgressBar.Horizontal"
-            android:layout_width="match_parent"
+            android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:layout_below="@id/home_toolbar"
             android:background="@color/vector_green_color"
             android:indeterminate="true"
-            android:visibility="gone" />
+            android:visibility="gone"
+            app:layout_constraintTop_toBottomOf="@+id/home_toolbar"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"/>
 
         <im.vector.view.VectorPendingCallView
             android:id="@+id/listView_pending_callview"
-            android:layout_width="match_parent"
+            android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:layout_below="@id/home_recents_sync_in_progress"
-            android:visibility="gone" />
+            android:visibility="gone"
+            app:layout_constraintLeft_toLeftOf="parent"
+            app:layout_constraintRight_toRightOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/home_recents_sync_in_progress" />
 
         <!-- The main content view -->
-        <RelativeLayout
-            android:layout_width="match_parent"
-            android:layout_height="match_parent"
-            android:layout_above="@+id/bottom_navigation"
-            android:layout_below="@+id/listView_pending_callview"
-            tools:context="org.matrix.vector.activity.VectorHomeActivity">
-
-            <FrameLayout
-                android:id="@+id/fragment_container"
-                android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:background="?attr/riot_primary_background_color" />
+        <android.support.constraint.Barrier
+            android:id="@+id/main_content_barrier"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:barrierDirection="bottom"
+            app:constraint_referenced_ids="listView_pending_callview"/>
```

Comment:
A Barrier is used to reference several ids. There is only one here, so I do not think this barrier is necessary. Maybe for code clarity?

Reply:
I added a barrier for clarity, I felt directly referencing the call view was clumsy but I've removed barrier as it is doing nothing.